### ISSUE TITLE
How to stop after x retries?

### ISSUE THREAD

**Issue by tasansga** [criteo/consul-templaterb#59]
If running in --once mode with invalid CONSUL_ADDR/CONSUL_TOKEN/VAULT_ADDR/VAULT_TOKEN, consul-templaterb will apparently error forever.

By comparison, consul-template has --retry and --vault-retry-attempts to stop early.

Is there a way to stop consul-templaterb after x errorenous retries? hot-reload has no effect (which makes sense since it's for the template, not the backend).

**Comment by pierresouchay** [comment]
It is currently not implemented, but it would be fairly easy to do so.
do you have ideas about implementation? Default values for vault/consul?
in which specific use case did you had this issue?

**Comment by tasansga** [comment]
My use-case is to run consul-templaterb exactly once to fill a template, and have it fail more or less immediately on any errors it encounters. I believe there's a default value for VAULT_ADDR and CONSUL_ADDR, because if it's set to empty I get this:


```
consul-templaterb --log-level=error --once --template templates/default.erb
-t --template templates/default.erb : Since output has not been set, using templates/default
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 11s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 11s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 12s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 12s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 14s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 14s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 18s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 18s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 26s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 26s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 42s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 42s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 74s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 74s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 138s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 138s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 266s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 266s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 522s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 522s
[ERROR] [v1/auth/token/renew-self][POST] Code: 0 Errno::ECONNREFUSED - Retry in 1034s
[ERROR] [/v1/secret/default][GET] Code: 0 Errno::ECONNREFUSED - Retry in 1034s
[ERROR] [/v1/][GET] Code: 0 Errno::ECONNREFUSED - Retry in 11s
[ERROR] [/v1/][GET] Code: 0 Errno::ECONNREFUSED - Retry in 12s
[ERROR] [/v1/][GET] Code: 0 Errno::ECONNREFUSED - Retry in 14s
[ERROR] [/v1/][GET] Code: 0 Errno::ECONNREFUSED - Retry in 18s
[ERROR] [/v1/][GET] Code: 0 Errno::ECONNREFUSED - Retry in 26s
[ERROR] [/v1/][GET] Code: 0 Errno::ECONNREFUSED - Retry in 42s
[ERROR] [/v1/][GET] Code: 0 Errno::ECONNREFUSED - Retry in 74s
[ERROR] [/v1/][GET] Code: 0 Errno::ECONNREFUSED - Retry in 138s
^C[WARN] consul-templaterb has been interrupted 
[KILL] received EXIT, stopping myself
```

Result after 1034s looks weird too, as if it reset?

**Comment by pierresouchay** [comment]
Yes, the values are CONSUL_HTTP_ADDR and VAULT_HTTP_ADDR (or you can specify it using flags).

In this case, that's because Vault address is not set properly. I'll add the option to allow stopping quickly in that case.

Do you have preferred option flag names?

**Comment by tasansga** [comment]
Ideas? Umm. Something like `--stop-on-failure=true|false    Immediately exit on failure? Default: false`
sounds simple, would solve my use-case and not change current default behaviour.

**Comment by pierresouchay** [comment]
@tasansga implemented in https://github.com/criteo/consul-templaterb/commit/80a7fe0edde2dfac1a6f72421b30fab6ee3c5b49 => does it sounds right for you?

**Comment by pierresouchay** [comment]
Implemented in version 1.23.0

**Comment by tasansga** [comment]
Awesome and quick work, thanks a lot!
